wicket-core/src/main/java/org/apache/wicket/Component.java

```diff
@@ -2478,7 +2478,8 @@ protected final void internalRenderComponent()
 				if (getFlag(FLAG_OUTPUT_MARKUP_ID))
 				{
 					String message = String.format("Markup id set on a component that renders its body only. " +
-					                               "Markup id: %s, component id: %s.", getMarkupId(), getId());
+					                               "Markup id: %s, component id: %s, class-type: %s, parent markup id: %s, parent relative path: %s, url: %s",
+							getMarkupId(), getId(), getClass(), getParent().getMarkupId(), getParent().getClassRelativePath(), getRequest().getOriginalUrl());
```

Comment:
1. I don't think we need the url.
2. Instead of ` getParent().getMarkupId(), getParent().getClassRelativePath()` could we use ` getClassRelativePath()`, i.e. `this.getClassRelativePath()` ? It would be simpler